mcu/apollo4p/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+# Ambiq HAL
+#
+# Copyright (c) 2023 Antmicro Ltd <www.antmicro.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Sources and headers necessary for every build.
+# These contain definitions and implementation used mostly for
+# initializing the SoC, and therefore are always required.
+zephyr_sources(hal/am_hal_global.c)
```

Comment:
any special reason why this file and those below are all added directly to the general Zephyr lib, and not in a dedicated hal lib ?

For example as:
```
zephyr_library()
zephyr_library_sources(...)
```

Reply:
There is no particular reason. I've changed it, thanks for the suggestion!